libs/kafka/logging.py

```diff
@@ -48,19 +48,21 @@ class LogTyp(Enum):
         INFO = ("blue", "[i]", "INFO")
         WARNING = ("yellow", "[!]", "WARNING")
 
-    def __init__(self, message, typ, servicename, mute=False):
+    def __init__(self, message, typ, servicename, functionname, mute=False):
         '''
         __init__ is the CTor for the LogMessage object.
         @param message is the message to log.
         @param typ will be a object from LogMessage.LogTyp
         @param servicename is the name of the service who logged
             the message.
+        @param functionname is the name of the function, the logging happens.
         @param mute is a boolean indicating whether a LogMessages
             should print a message.
         '''
         self.message = message
         self.typ = typ
         self.servicename = servicename
+        self.functionname = functionname
```

Comment:
Instead of passing the function name as param I would extract it from the stack. That should work generic and we don't need to change everything. Maybe [that](https://stackoverflow.com/questions/2654113/how-to-get-the-callers-method-name-in-the-called-method) works?

Reply:
Done.